GoogleForCreators/web-stories-wp
# Re-index stories in Yoast SEO if permalinks are outdated

## Summary

We changed the rewrite slug from `/stories/` to `/web-stories/` mid-way.

Because Yoast SEO caches the permalinks in the database, it means stories created prior to that change will have outdated canonical URLs and the like

## Relevant Technical Choices

Usage of all these Yoast SEO classes required PHPStan stubs. Took the time to improve the PHPStan config using the recommendations at https://phpstan.org/user-guide/discovering-symbols.

Unfortunately https://github.com/php-stubs/wordpress-seo-stubs is outdated, otherwise we could have used that package directly for the stubs.

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

1. Activate Yoast SEO
1. Install Web Stories for WordPress Beta 2
1. Publish a story
1. Update to 1.0.0 stable
1. Check story on the frontend, notice the **old** URLs
1. Switch to this branch
1. Visit WordPress admin to trigger the db upgrader
1. Check story on the frontend, notice the **new** URLs

---



Fixes #4754
